pkg/pool/pool.go

```diff
@@ -40,36 +46,41 @@ func NewBytesPool(minSize, maxSize int, factor float64) *BytesPool {
 		sizes = append(sizes, s)
 	}
 
-	p := &BytesPool{
+	p := &Pool{
 		buckets: make([]sync.Pool, len(sizes)),
 		sizes:   sizes,
+		make:    makeFunc,
 	}
 
 	return p
 }
 
 // Get returns a new byte slices that fits the given size.
-func (p *BytesPool) Get(sz int) []byte {
+func (p *Pool) Get(sz int) interface{} {
 	for i, bktSize := range p.sizes {
 		if sz > bktSize {
 			continue
 		}
-		b, ok := p.buckets[i].Get().([]byte)
-		if !ok {
-			b = make([]byte, 0, bktSize)
+		b := p.buckets[i].Get()
+		if b == nil {
+			b = p.make(bktSize)
 		}
 		return b
 	}
-	return make([]byte, 0, sz)
+	return p.make(sz)
 }
 
-// Put returns a byte slice to the right bucket in the pool.
-func (p *BytesPool) Put(b []byte) {
-	for i, bktSize := range p.sizes {
-		if cap(b) > bktSize {
+// Put adds a slice to the right bucket in the pool.
+func (p *Pool) Put(s interface{}) {
+	slice := reflect.ValueOf(s)
+
+	if slice.Kind() != reflect.Slice {
+		panic(fmt.Sprintf("%+v is not a slice", slice))
+	}
+	for i, size := range p.sizes {
+		if slice.Cap() > size {
 			continue
 		}
-		p.buckets[i].Put(b[:0])
-		return
+		p.buckets[i].Put(slice.Slice(0, 0).Interface())
```

Comment:
You dropped the early `return` – this won't end well ;)

Reply:
oops :shamrock: